airconsole-1.9.0-alpha.js

```diff
@@ -1092,11 +1129,65 @@ AirConsole.prototype.init_ = function(opts) {
     version: me.version,
     device_motion: opts.device_motion,
     synchronize_time: opts.synchronize_time,
+    silence_players: me.silence_players,
     location: me.getLocationUrl_(),
     translation: opts.translation
   });
+  if(me.silence_players) {
+     this.enablePlayerSilencing_(me.silence_players);
+  }
 };
 
+/**
+ * Enables the player silencing
+ * @param {boolean} silence_players If true, the player silencing feature is active
+ * @private
+ * @since 1.9.0
+ */
+AirConsole.prototype.enablePlayerSilencing_ = function (silence_players) {
+  if(silence_players === undefined || this.silence_players === silence_players) {
+    return;
+  }
+
+  this.set_("silence_players", silence_players);
+}
+
+/**
+ * Checks if the location is in the same location of the sender
+ * @param {string} location The location to check.
+ * @param {string} sender_id The id of the sender.
+ * @returns {boolean} True if the location are identical.
+ * @private
+ */
+AirConsole.prototype.isDeviceInSameLocation_ = function (location, sender_id) {
+  return !!this.devices[sender_id] && location === this.getGameUrl_(this.devices[sender_id].location);
+}
+
+/**
+ * Queries if a given device_id is a player in the current session and not silenced.
+ * @param {string} device_id The device_id to query the information for
+ * @returns {boolean} True, if the receiver with device_id is silenced.
+ * @private
+ */
+AirConsole.prototype.receiverIsSilenced_ = function (device_id) {
+  return this.arePlayersSilenced() && device_id !== undefined
+    && device_id !== AirConsole.SCREEN
+    && this.convertDeviceIdToPlayerNumber(device_id) === undefined;
+}
+
+/**
+ * Queries if a given device_id is silenced
+ * @param {string} device_id  The device_id to be queried.
+ * @returns {boolean} True, if the sender with device_id is silenced.
+ * @private
+ */
+AirConsole.prototype.senderIsSilenced_ = function (device_id) {
```

Comment:
Since senderIsSilenced_ and receiverIsSilenced_ are the exact same function, I would recommend to join it to deviceIsSilenced_

Reply:
You are right, thank you for catching this.
I missed that the the senderIsSilenced had become identicial over the necessary improvement when integrating it in the test game.